tests/integration/test_block.py

```diff
@@ -50,31 +55,25 @@ def run_test(arr, input_sizes, depth):
     )
 
 
-def test(dim):
-    print("test np.block")
-    # test append w/ 1D, 2D and 3D arrays
-    input_options = [
-        ((0,), (0,)),  # empty arrays
-        ((1,), (1,)),  # singlton arrays
-        ((dim, 1), (dim, dim)),  # 1D and 2D arrays
-        ((dim, 1), (dim, 1), (dim, dim)),  # 3 arrays in the inner-most list
-    ]
-    for input_sizes in input_options:
-        for depth in range(2):  # test depth from 1 to 3
-            a = list(
-                np.arange(np.prod(input_size)).reshape(input_size)
-                for input_size in input_sizes
-            )
-            b = list(
-                np.arange(np.prod(input_size)).reshape(input_size)
-                for input_size in input_sizes
-            )
-            for nlist in range(depth):
-                a = [a]
-                b = [b]
-            run_test([a, b], input_sizes, depth)
-    return
+DIM = 10
+
+SIZE_CASES = [
+    [(0,), (0,)],  # empty arrays
+    [(1,), (1,)],  # singlton arrays
+    [(DIM, 1), (DIM, DIM)],  # 1D and 2D arrays
+    [(DIM, 1), (DIM, 1), (DIM, DIM)],  # 3 arrays in the inner-most list
+]
+
+
+@pytest.mark.parametrize("sizes", SIZE_CASES, ids=str)
+@pytest.mark.parametrize("depth", range(3))
```

Comment:
The original test stated "test depth from 1 to 3" but only had `range(2)`. The tests pass up to `range(3)` though, so I upped it here.